### PR TITLE
AC: fix paths conversion

### PR DESCRIPTION
**Pull request by eaidova**
fix for relative path in config without prefix usage

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/annotation_converters/style_transfer.py [hunk 1]
```diff
@@ -35,7 +36,7 @@ def parameters(cls):
         return parameters
 
     def configure(self):
-        self.image_dir = self.get_value_from_config('images_dir')
+        self.image_dir = Path(self.get_value_from_config('images_dir'))
```
Comment (IRDonch) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/style_transfer.py:39]: Why is this needed? I thought `images_dir` being a `PathField` meant that the value would already be a `Path`.

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/style_transfer.py]: you are right, I deeper investigate issue and found that type casting never performend during getting value from config. native yml read it as string. when we use path without  prefixing in the config it saved as string

File: tools/accuracy_checker/accuracy_checker/config/config_validator.py [hunk 2]
```diff
@@ -152,7 +152,11 @@ def validate(self, entry, field_uri=None):
 
     @property
     def type(self):
-        return str
+        def type_def(x=None):
+            if x is None:
+                return str
+            return x
```
Comment (IRDonch) [inline tools/accuracy_checker/accuracy_checker/config/config_validator.py:158]: I don't understand what's going on here. This function returns `str` (a type) if `x` is `None`, but `x` (a value) otherwise. That doesn't seem right. And why is it needed at all? The logic in `utils.py` already handles the case when `x` is `None`.

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/config/config_validator.py:158]: getting type from Field also used for static analisys required configuration for model in DL WorkBench. They want to get field type without providing argument. The BaseField used in cases, where several types applicable (e.g. for mean, we can provide float, list of floats for each channel or select one of precomputed values using strings), we can not know which type will have input values, but during parsing is opportunity to parsing necessary formats from string always exists.

Comment (IRDonch) [reply tools/accuracy_checker/accuracy_checker/config/config_validator.py:158]: I see; however, your code doesn't seem to actually support this use case: if `f` is an instance of `BaseField`, then `f.type` is `type_def`, not `str`.

IMO, a better way to do this would be to have `type` return `None` here, and add special logic for `None` both in `get_parameter_value_from_config` and in DL WB.

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/config/config_validator.py]: ok, changes

File: tools/accuracy_checker/accuracy_checker/config/config_validator.py [hunk 3]
```diff
@@ -165,7 +165,12 @@ def parameters(self):
                     parameters_dict[key] = self.__dict__[key].parameters()
                 else:
                     parameters_dict[key] = self.__dict__[key]
-            parameters_dict['type'] = type(self.type()).__name__
+            data_type = self.type
+            if data_type is not None:
+                data_type = type(data_type()).__name__  # pylint:disable=E1102
+            else:
+                data_type = str.__name__
+            parameters_dict['type'] = data_type
```
Comment (IRDonch) [inline tools/accuracy_checker/accuracy_checker/config/config_validator.py:173]:
```suggestion
            parameters_dict['type'] = type((self.type or str)()).__name__
```
